metrics.py

```diff
@@ -427,17 +439,25 @@ def calculate_sample_statistics(self, networks_sample: np.ndarray) -> np.ndarray
         if self.requires_graph:
             networks_as_graphs = [connectivity_matrix_to_G(W, self.is_directed) for W in networks_sample]
 
+        networks_as_sparse_tensor = np_tensor_to_sparse_tensor(networks_sample)
```

Comment:
We can get rid of the `numpy` networks sample after converting it to a sparse tensor (I don't know if this solves the memory leakage problem, but worth trying if we just try deleting stuff here and there).